python/core/auto_generated/qgsfeaturerequest.sip.in

```diff
@@ -736,6 +736,21 @@ For example, this should be set on requests that are issued from an
 expression function.
 
 .. versionadded:: 3.4
+%End
+
+    void stackFilters( bool useFids );
```

Comment:
I think this could be reworded for clarity. A better (more explicit and flexible) approach could be:

    enum FilterCombinationMethod
    {
        MethodReplace, //!< Removes any existing feature ID filter when a filter expression is set, or vice-versa. This is the default behavior.
        OperatorAnd, //!< Filter expression and feature ID filters are combined using a logical "AND" operation, so that features must satisfy both the ID filter and the expression filter
        // TODO: OperatorOr,
     }

     void setCombinationMethod( CombinationMethod method );

     CombinationMethod combinationMethod() const;

Reply:
I'll see what I can do, I'm not too at ease with your proposed approach at the moment. I'll try to come up with an alternate solution.

Reply:
@nyalldawson opinion on the new method/overhaul? I know this is not what you had in mind but this should cover all cases and be extendable while providing a simpler check.